### ISSUE TITLE
Upgrading az MSI installation in non-interactive way

### ISSUE THREAD

**Issue by nascarsayan** [Azure/azure-cli#27456]
**Related command**

`az upgrade`

**Is your feature request related to a problem? Please describe.**

When azure CLI is installed via MSI in windows, there is no method of automated upgrade using script. `az upgrade` will invoke `msiexec.exe` which opens up a separate window, and the user has to follow the steps (by clicking Next, etc.).

**Describe the solution you'd like**

`az upgrade` might take in a parameter `--yes` for automated MSI installation of the latest version. Else, we can make automated installation the default / only behaviour.

Basically, I propose that we need to have an automated upgrade mechanism without user intervention.

**Describe alternatives you've considered**

From our installation / onboarding script, I manually check the latest version available by parsing `setup.py`, similar to how `az upgrade` code snippet does. If the installed version is not equal to the latest version, I download and install the MSI in non-interactive way. 
This roundabout method can be replaced by `az upgrade` if there is a non-interactive way for `az upgrade`.

**Additional context**

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by jiasli** [comment]
As far as I know, there are some technical difficulties preventing us from doing this.

As shown in https://github.com/Azure/azure-cli/pull/26464, the MSI installation GUI shows a warning if `python.exe` is occupied but needs to be updated, so we want to make `az upgrade` non-blocking in that PR. With the MSI installation GUI, the user can still know if the MSI installation has finished. But if we make the MSI installation non-interactive and `az upgrade` non-blocking by exiting the current `python.exe` process, there will be no way to know if the MSI installation has finished (succeeded/failed).

A workaround I can currently think of is to **develop a separate script independent from Azure CLI**, so that it can upgrade Azure CLI without launching Azure CLI.
